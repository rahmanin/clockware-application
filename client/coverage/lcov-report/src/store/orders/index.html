
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/store/orders</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/store/orders</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.64% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>39/44</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.92% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>10/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.92% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90.24% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>37/41</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="actions.ts"><a href="actions.ts.html">actions.ts</a></td>
	<td data-value="73.68" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 73%"></div><div class="cover-empty" style="width: 27%"></div></div>
	</td>
	<td data-value="73.68" class="pct medium">73.68%</td>
	<td data-value="19" class="abs medium">14/19</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="18" class="abs medium">14/18</td>
	</tr>

<tr>
	<td class="file high" data-value="reducer.ts"><a href="reducer.ts.html">reducer.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="81.82" class="pct high">81.82%</td>
	<td data-value="11" class="abs high">9/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Tue Dec 01 2020 12:22:24 GMT+0200 (Eastern European Standard Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    