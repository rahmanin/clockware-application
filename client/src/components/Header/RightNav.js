import React, {useEffect} from 'react';
import styled from 'styled-components';
import {Link} from 'react-router-dom';
import { routes } from "../../constants/routes";
import { useHistory } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import {userParams} from "../../store/users/selectors";
import {logOut} from "../../store/users/actions";

const Ul = styled.ul`
  list-style: none;
  display: flex;
  flex-flow: row nowrap;
  li {
    padding: 18px 10px;
  }
  z-index: 1;
  @media (max-width: 768px) {
    flex-flow: column nowrap;
    background-color: rgba(209, 209, 209, 0.9);
    position: fixed;
    transform: ${({ open }) => open ? 'translateX(0)' : 'translateX(100%)'};
    top: 0;
    right: 0;
    height: 100vh;
    width: 300px;
    padding-top: 3.5rem;
    transition: transform 0.3s ease-in-out;
    li {
      color: #fff;
    }
  }
`;

const RightNav = ({ open, onClick }) => {
  const userData = useSelector(userParams);
  const history = useHistory();
  const dispatch = useDispatch();

  const logIn = () => {
    userData && userData.userId ? (history.push(routes.order) || localStorage.clear() || dispatch(logOut())) : history.push(routes.login);
    onClick()
  }
  return (
    <Ul open={open}>
      <li hidden={!userData || !userData.userId || !userData.is_admin}>
        <Link to={routes.diagrams} onClick={onClick}>
          <div className="links">Diagrams</div>
        </Link>
      </li>
      <li hidden={!userData || !userData.userId || !userData.is_admin}>
        <Link to={routes.masters} onClick={onClick}>
          <div className="links">Masters</div>
        </Link>
      </li>
      <li hidden={!userData || !userData.userId || !userData.is_admin}>
        <Link to={routes.cities} onClick={onClick}>
          <div className="links">Cities</div>
        </Link> 
      </li>
      <li hidden={!userData || !userData.userId}>
        <Link to={routes.orders} onClick={onClick}>
          <div className="links" >Orders</div>
        </Link> 
      </li>
      <li hidden={!userData || !userData.userId || !userData.is_admin}>
        <Link to={routes.prices} onClick={onClick}>
          <div className="links" >Prices</div>
        </Link>
      </li>
      <li>
        <div className="links" onClick={logIn}>{userData && userData.userId ? "Log out" : "Log in"}</div>
      </li>
    </Ul>
  )
}

export default RightNav