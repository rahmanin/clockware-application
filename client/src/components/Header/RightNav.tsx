import React, {FunctionComponent} from 'react';
import styled from 'styled-components';
import {Link} from 'react-router-dom';
import { routes } from "../../constants/routes";
import { useSelector, useDispatch } from "react-redux";
import {userParams} from "../../store/users/selectors";
import {UserData} from "../../store/users/actions";
import { Menu, Dropdown } from 'antd';
import { DownOutlined } from '@ant-design/icons';
import { switchLanguage } from '../../store/language/actions';
import { useTranslation } from 'react-i18next';
import { currentLanguage } from '../../store/language/selectors';
import { LoginBtns } from './LoginBtns';

interface Props {
  open: boolean,
  onClick: () => void
}

const Ul = styled.ul`
  list-style: none;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  margin: 0;
  .language_menu {
    display: flex;
    align-items: center;
    >* {
      margin-left: 10px;
    }
  }
  li {
    margin: 0 10px 0 10px;
  }
  .login_btn, .sign_up_btn {
    display: none
  }
  z-index: 9999;
  @media (max-width: 1080px) {
    align-items: flex-start;
    flex-flow: column nowrap;
    background: #F2F2F2;
    position: fixed;
    transform: ${({ open }: {open: boolean}) => open ? 'translateX(0)' : 'translateX(100%)'};
    top: 0;
    right: 0;
    height: 100vh;
    width: 300px;
    padding-top: 3.5rem;
    transition: transform 0.3s ease-in-out;
    li {
      color: #fff;
      margin: 14px;
    }
    .login_btn, .sign_up_btn {
      display: block
    }
  }
`;

export const RightNav: FunctionComponent<Props> = ({ open, onClick }) => {
  const language: string = useSelector(currentLanguage)
  const userData: UserData = useSelector(userParams);
  const dispatch: Function = useDispatch();
  const isAdmin: boolean = userData && userData.role === "admin";
  const { t } = useTranslation('common')

  const languageMenu = (
    <Menu>
      <Menu.Item 
        key="0" 
        onClick={() => dispatch(switchLanguage("en"))} 
        className={language === "en" ? "selected_language" : ""}
      >
        <img className="country_flag" src="http://purecatamphetamine.github.io/country-flag-icons/3x2/US.svg"/>
      </Menu.Item>
      <Menu.Item 
        key="1" 
        onClick={() => dispatch(switchLanguage("ru"))}
        className={language === "ru" ? "selected_language" : ""}
      >
        <img className="country_flag" src="http://purecatamphetamine.github.io/country-flag-icons/3x2/RU.svg"/>
      </Menu.Item>
    </Menu>
  );

  return (
    <Ul open={open}>
      <li hidden={!userData || !isAdmin}>
        <Link to={routes.calendar} onClick={onClick}>
          <div className="links">{t("Header.Calendar")}</div>
        </Link>
      </li>
      <li hidden={!userData || !isAdmin}>
        <Link to={routes.diagrams} onClick={onClick}>
          <div className="links">{t("Header.Statistics")}</div>
        </Link>
      </li>
      <li hidden={!userData || !isAdmin}>
        <Link to={routes.masters} onClick={onClick}>
          <div className="links">{t("Header.Masters")}</div>
        </Link>
      </li>
      <li hidden={!userData || !isAdmin}>
        <Link to={routes.cities} onClick={onClick}>
          <div className="links">{t("Header.Cities")}</div>
        </Link> 
      </li>
      <li hidden={!userData || !userData.role}>
        <Link to={routes.orders} onClick={onClick}>
          <div className="links">{t("Header.Orders")}</div>
        </Link> 
      </li>
      <li hidden={!userData || !isAdmin}>
        <Link to={routes.prices} onClick={onClick}>
          <div className="links">{t("Header.Prices")}</div>
        </Link>
      </li>
      <li>
        <Link to={routes.blog} onClick={onClick}>
          <div className="links">{t("Header.Blog")}</div>
        </Link>
      </li>
      <li hidden={!userData || !isAdmin}>
        <Link to={routes.map} onClick={onClick}>
          <div className="links">{t("Header.Map")}</div>
        </Link>
      </li>
      <li>
        <div className="links">
        <Dropdown overlay={languageMenu} trigger={['click']}>
          <div className="language_menu">
            {t("Header.Language")} <DownOutlined/>
          </div>
        </Dropdown>
        </div>
      </li>
      <li>
      <Link to={routes.about} onClick={onClick}>
        <div className="links">{t("Header.About")}</div>
      </Link>
      </li>
      <li>
        <LoginBtns classNameSignIn="login_btn" classNameSignUp="sign_up_btn" onClick={onClick}/>
      </li>
    </Ul>
  )
}