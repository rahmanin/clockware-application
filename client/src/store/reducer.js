import { combineReducers } from "redux";
import pricesReducer from "./prices/reducer";

const rootReducer = combineReducers({
  prices: pricesReducer,
});

export default rootReducer;