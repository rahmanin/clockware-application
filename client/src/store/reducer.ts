import { combineReducers } from "redux";
import pricesReducer from "./prices/reducer";
import usersReducer from "./users/reducer";
import citiesReducer from "./cities/reducer";
import mastersReducer from "./masters/reducer";
import ordersReducer from "./orders/reducer";
import ordersClientReducer from "./ordersClient/reducer";
import languageReducer from "./language/reducer";
import { createStore, applyMiddleware } from "redux";
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension'

const rootReducer = combineReducers({
  prices: pricesReducer,
  user: usersReducer,
  cities: citiesReducer,
  masters: mastersReducer,
  orders: ordersReducer,
  ordersClient: ordersClientReducer,
  language: languageReducer
});

const composedEnhancer = composeWithDevTools(
  applyMiddleware(thunk)
)

const store = createStore(
  rootReducer,
  composedEnhancer
);

export default store
