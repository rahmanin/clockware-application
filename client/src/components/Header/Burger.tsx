import React, { useState, FunctionComponent } from "react";
import styled, { StyledComponent } from "styled-components";
import {RightNav} from "./RightNav";
import { LoginBtns } from './LoginBtns';

interface Prop {
  open: boolean
}

const StyledBurger: StyledComponent<"div", any, Prop, never> = styled.div`
  width: 2rem;
  height: 2rem;
  position: fixed;
  top: 14px;
  right: 20px;
  z-index: 99999;
  display: none;
  @media (max-width: 1080px) {
    display: flex;
    justify-content: space-around;
    flex-flow: column nowrap;
  }
  div {
    width: 2rem;
    height: 0.25rem;
    background-color: #333;
    border-radius: 10px;
    transform-origin: 1px;
    transition: all 0.3s linear;
    &:nth-child(1) {
      transform: ${({ open }: Prop) => (open ? "rotate(45deg)" : "rotate(0)")};
    }
    &:nth-child(2) {
      transform: ${({ open }: Prop) => (open ? "translateX(100%)" : "translateX(0)")};
      opacity: ${({ open }: Prop) => (open ? 0 : 1)};
    }
    &:nth-child(3) {
      transform: ${({ open }: Prop) => (open ? "rotate(-45deg)" : "rotate(0)")};
    }
  }
`;

export const Burger: FunctionComponent = () => {
  const [open, setOpen] = useState<boolean>(false);

  return (
    <>
      <StyledBurger open={open} onClick={() => setOpen(!open)}>
        <div />
        <div />
        <div />
      </StyledBurger>
      <RightNav open={open} onClick={() => setOpen(false)} />
      <LoginBtns classNameSignIn="login_btn hide" classNameSignUp="sign_up_btn hide" onClick={() => setOpen(false)}/>
    </>
  );
};
