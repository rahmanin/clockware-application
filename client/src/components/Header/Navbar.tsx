import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import {Burger} from './Burger';
import { routes } from "../../constants/routes";
import {Link} from 'react-router-dom';

const Nav = styled.nav`
  display: flex;
  justify-content: space-between;
  .logo {
    margin-left: 5px;
    font-size: 40px;
    color: rgb(0, 0, 0);
  }
`

export const Navbar: FunctionComponent = () => {
  return (
    <div className="header_background">
      <Nav className="header_wrapper">
        <Link to={routes.order} className="header_logo">
          <div className="logo_png"></div>
          <p>Clockware</p>
        </Link>
        <Burger />
      </Nav>
    </div>
  )
}