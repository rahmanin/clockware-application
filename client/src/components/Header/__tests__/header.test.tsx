import {Header} from "../index"
import {Burger} from "../Burger"
import {Navbar} from "../Navbar"
import {RightNav} from "../RightNav"
import React from 'react';
import {cleanup} from '@testing-library/react';
import { Provider } from "react-redux";
import store from "../../../store/reducer";
import { act } from "react-dom/test-utils";
import {shallow} from "enzyme"
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });
afterEach(cleanup);

act(() => {
  it('Render Header', () => {
    const component = shallow(
      <Header />
    );
    expect(component).toMatchSnapshot();
  });
})

act(() => {
  it('Render Burger', () => {
    const component = shallow(
      <Burger />
    );
    expect(component).toMatchSnapshot();
  });
})

act(() => {
  it('Render Navbar', () => {
    const component = shallow(
      <Navbar />
    );
    expect(component).toMatchSnapshot();
  });
})

act(() => {
  it('Render RightNav', () => {
    const component = shallow(
      <Provider store={store}>
        <RightNav open={true} onClick={() => {return}}/>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
})