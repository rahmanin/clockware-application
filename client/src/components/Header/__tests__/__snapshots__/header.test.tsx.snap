// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Burger 1`] = `
<Fragment>
  <styled.div
    onClick={[Function]}
    open={false}
  >
    <div />
    <div />
    <div />
  </styled.div>
  <RightNav
    onClick={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`Render Header 1`] = `
<div
  className="header"
  id="header"
>
  <Navbar />
</div>
`;

exports[`Render Navbar 1`] = `
<styled.nav>
  <Link
    to="/order"
  >
    <div
      className="logo"
    >
      CLOCKWARE
    </div>
  </Link>
  <Burger />
</styled.nav>
`;

exports[`Render RightNav 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <RightNav
    onClick={[Function]}
    open={true}
  />
</ContextProvider>
`;
